aspnetcore/fundamentals/routing.md

```diff
@@ -135,6 +135,16 @@ Routing is connected to the [middleware](xref:fundamentals/middleware/index) pip
 
 URL matching is the process by which routing dispatches an incoming request to an *endpoint*. This process is based on data in the URL path but can be extended to consider any data in the request. The ability to dispatch requests to separate handlers is key to scaling the size and complexity of an app.
 
+::: moniker-end
+
+::: moniker range=">= aspnetcore-3.0"
+
+When a Routing Middleware executes, it sets an endpoint (`Endpoint`) and route values to a feature on the <xref:Microsoft.AspNetCore.Http.HttpContext>. Middleware running after the Routing Middleware can see the endpoint and can take actions based on the middleware. For example, an Authorization Middleware can interrogate the endpoint's metadata collection for an authorization policy. After all of the middleware in the request processing pipeline is executed, the selected endpoint's delegate is invoked.
```

Comment:
I think this should be expanded to include how to get the endpoint/route values after they have been set.

* HttpContext.GetEndpoint gets the endpoint - https://github.com/aspnet/AspNetCore/blob/f45250ce5e423dad083fe99967a4d65782c5301e/src/Http/Http.Abstractions/src/Routing/EndpointHttpContextExtensions.cs#L19
* HttpRequest.RouteValues gets the route values - https://github.com/aspnet/AspNetCore/blob/f45250ce5e423dad083fe99967a4d65782c5301e/src/Http/Http.Abstractions/src/HttpRequest.cs#L132